### DISCUSSION TITLE
Livewire defer with select2 not working

### DISCUSSION THREAD

**Discussion by Vidhi-Aecordigital** [livewire/livewire#2509] [1. Help]
As per the documentation of livewire, wire:model.defer (defer updating) seems to work only for input type. However it also work for normal select but not for select2 . Is there any work around to use wire:model.defer for select2. 

https://laravel-livewire.com/docs/2.x/properties

**Comment by callmevladik** [comment]
@Vidhi-Aecordigital in your select2 options **on select event** you can dispatchEvent(new Event('change')). It worked for me with the form that needs to be saved. Select has wire:model.defer and the request is being send only after saving the form

**Comment by ousid** [comment]
Hey, @Vidhi-Aecordigital 

No need to defer in this case. Here is how I solved the issue my self with `select2`

```blade
<select id="users-dropdown">
    @foreach($users as $user)
        <option value="{{ $user->id }}" wire:key="{{ $user->id }}" >{{ $user->name] }}</option>
    @endforeach
</select>
```

```js
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>
<script>
    document.addEventListener("DOMContentLoaded", () => {
        $('#users-dropdown').select2();

        $('#users-dropdown').on('change', function (e) {
            var data = $('#users-dropdown').select2("val");
            @this.set('userProp', data);
        });

        window.addEventListener('contentChanged', event => {

            $('#users-dropdown').on('change', function (e) {
                var data = $('#users-dropdown').select2("val");
                @this.set('userProp', data);
            });

        });
    });
</script>
```

**Comment by amiranagram** [comment]
Hi. if you need select2 multi try this one out. Modify it to suit your own needs.

https://gist.github.com/amiranagram/896d17e92c2c42e4ea9f59043f696bc6

Modify the `select2:select` and `select2:unselect` callbacks to adapt it to a non-multi select.

**Comment by skalero01** [comment]
I need help on this also :S

**Comment by hyphen81** [comment]
Has anyone figured out a way to use wire:model.defer with select2? I have a component where I'm doing the `@this.set(...` thing, but in one case I want to hold off on sending a value until the form is submitted, but when it's submitted the value of the variable is null.

**Comment by JeffGepiga** [comment]
> Has anyone figured out a way to use wire:model.defer with select2? I have a component where I'm doing the `@this.set(...` thing, but in one case I want to hold off on sending a value until the form is submitted, but when it's submitted the value of the variable is null.
```
$(document).on('select2:select', '#element',function({
  @this.set('variable',this.value,true)
})

or

<select onchange="@this.set('variable',this.value,true)"></select>
```
